### PR TITLE
Windows Appveyor Build

### PR DESCRIPTION
**Pull request by JeffAshton**
Creating an appveyor.yml build file so that telegraf can be compiled and tested under windows for CI.  This is intended to supplement the existing CircleCI pipeline.

Corresponding issue https://github.com/influxdata/telegraf/issues/1719

### Required for all PRs:

- [Y] Signed [CLA](https://influxdata.com/community/cla/).
- [N/A] Associated README.md updated.
  - Changes aren't functional
- [N/A] Has appropriate unit tests.
  - Runs existing unit tests

### CODE REVIEW

**Comment by JeffAshton** [comment]
The result of this PR in appveyor:

https://ci.appveyor.com/project/Brightspace/telegraf/build/1.4-rc103

**Comment by JeffAshton** [comment]
The version property is mostly only information:

image

image

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+clone_folder: C:\gopath\src\github.com\influxdata\telegraf
+
+environment:
+  GOPATH: C:\gopath
+
+platform: x64
+
+install:
+  - IF NOT EXIST "C:\Cache" mkdir C:\Cache
+  - IF NOT EXIST "C:\Cache\go1.8.1.msi" curl -o "C:\Cache\go1.8.1.msi" https://storage.googleapis.com/golang/go1.8.1.windows-amd64.msi
```
Comment (JeffAshton) [inline appveyor.yml]: Same version of GO as circle CI build

**Comment by danielnelson** [comment]
Merging into a branch for more testing.

**Comment by danielnelson** [comment]
Not sure what to do about this, AppVeyor seems to want permission to modify all of my public repos if I create the project as a github project.  Docs seem to confirm this:

>For every project AppVeyor will configure webhooks for its repository to automatically start a build when you push the changes. For every private project AppVeyor will add an SSH public key (deployment key) to the clone repository on the build machine.

I have admin access to many public repos, and I can't have them all getting modified.  I tried setting the repo up as a plain git project but it doesn't integrate very well.  @JeffAshton do you know if it is going to modify all of my repos if I do the github integration?

**Comment by danielnelson** [comment]
Okay, I got gutsy and luckily it just asks for permission to all so that it can list the repos.

**Comment by JeffAshton** [comment]
Sorry for the late response. My experience has been the same as you discovered.  We have lots of projects, but only once you add them to appveyor do they add the hooks.  Our experience is that it's been very similar to CircleCI.

Thank you @danielnelson for taking this one.  Greatly appreciated by all the Windows users I'm sure.  Should give everyone a lot more confidence moving forward.

**Pull request merged**